core/src/main/java/io/questdb/std/str/Utf8StringSink.java

```diff
@@ -106,13 +106,20 @@ public Utf8StringSink put(byte b) {
         return putByte0(b);
     }
 
+    @Override
+    public Utf8Sink putAny(byte b) {
+        ascii &= b >= 0;
+        return putByte0(b);
+    }
+
     @Override
     public Utf8Sink putAscii(char c) {
+        assert c < 128 : "c is non-ASCII";
```

Comment:
nit: this may actually have some perf impact since we enable assertions in -rt artifacts.